source/tutorial/kotlin.txt

```diff
@@ -298,47 +423,59 @@ Change the Subscription
       a value of 3. We can make direct comparisons between an integer and the 
-      Priority property. To do so, edit the RQL statement to include documents 
+      priority property. To do so, edit the RQL statement to include documents 
       where the priority is equal to or *less* than PriorityLevel.High (or 1), as 
       shown here:
 
       .. code-block:: kotlin
-         :emphasize-lines: 4-6
-
-         config = SyncConfiguration.Builder(user, setOf(Item::class))
-             .initialSubscriptions (rerunOnOpen = true) { realm ->
-                 add(
-                     realm.query<Item>(
-                         "owner_id == $0 AND priority <= ${PriorityLevel.High.priority.toString()}",
-                         user.identity
-                     ),
-                     "User's Items",
-                     updateExisting = true,
-                 )
-             }
-
-      Notice we've set ``rerunOnOpen`` to true so that the closure is re-run
-      every time the Realm is opened. We've also set ``updateExisting`` to
-      true, so the Realm Kotlin SDK updates the previous query named "User's
-      Item", rather than subscribing to two queries.
+         :emphasize-lines: 3
+
+         private fun getQuery(realm: Realm, subscriptionType: SubscriptionType): RealmQuery<Item> =
+            when (subscriptionType) {
+                  SubscriptionType.MINE -> realm.query("owner_id == $0 AND priority <= ${PriorityLevel.High.ordinal}", currentUser.id)
+                  SubscriptionType.ALL -> realm.query()
+            }
+      
+      We'll also force the subscription query to recalculate which documents to 
+      sync every time we open the app. 
+
+      To do this, find the ``SyncConfiguration.Builder().initialSubscriptions()`` function 
+      that our application calls on start, and add the ``reRunOnOpen`` parameter set to ``true``: 
+
+      .. code-block:: kotlin
+         :emphasize-lines: 2
+
+         config = SyncConfiguration.Builder(currentUser, setOf(Item::class))
+            .initialSubscriptions(rerunOnOpen = true) { realm ->
+                // Subscribe to the active subscriptionType - first time defaults to MINE
+                val activeSubscriptionType = getActiveSubscriptionType(realm)
+                add(getQuery(realm, activeSubscriptionType), activeSubscriptionType.name)
+            }
+            .errorHandler { session: SyncSession, error: SyncException ->
+                onSyncError.invoke(session, error)
+            }
+            .waitForInitialRemoteData()
+            .build()
 
    .. step:: Run and Test
     
       Run the application again. Log in using the account you created earlier in
       this tutorial. Here, you'll notice an error if you check your
-      :android:`logs using Logcat <studio/debug/am-logcat>`, you will see an
+      :android:`logs using Logcat <studio/debug/am-logcat>`. You will see an
```

Comment:
When I first ran this, I got a different error:

```
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{com.mongodb.app/com.mongodb.app.ComposeItemActivity}: java.lang.IllegalStateException: Existing query 'MINE' was found and could not be updated as `updateExisting = false`
```
After I wiped the app from the emulator and ran it again, it worked as expected. We might want to add something to the tutorial around dealing with this possibility.

Reply:
I added `updateExisting = true` as part of the code sample, which addresses this error. just tested and it works as expected (you get the documented error even if you don't wipe the app)